File: UiMetadataFramework.Core/ClientFunctionMetadata.cs
namespace UiMetadataFramework.Core
{
	using System.Collections.Generic;

	/// <summary>
	/// Represents a client-side function.
	/// </summary>
	public class ClientFunctionMetadata
	{
		/// <summary>
		/// Initializes a new instance of the <see cref="ClientFunctionMetadata"/> class.
		/// </summary>
		/// <param name="id"></param>
		public ClientFunctionMetadata(string id) : this(id, null)
		{
		}

		/// <summary>
		/// Initializes a new instance of the <see cref="ClientFunctionMetadata"/> class.
		/// </summary>
		/// <param name="id"></param>
		/// <param name="customProperties"></param>
		public ClientFunctionMetadata(string id, IDictionary<string, object> customProperties)
		{
			this.Id = id;
			this.CustomProperties = customProperties;
		}

		/// <summary>
		/// Gets or sets custom properties describing how to run the function.
		/// </summary>
		public IDictionary<string, object> CustomProperties { get; set; }

		/// <summary>Gets or sets id of the function.</summary>
		public string Id { get; }
	}
}

File: uimf-core/src/ClientFunctionMetadata.ts
/**
 * Represents a function.
 */
export class ClientFunctionMetadata {
    constructor(metadata: any) {
		for (var property of Object.keys(metadata)) {
			this[property] = metadata[property];
		}
    }

    /**
     * Gets or sets custom properties describing how to run the function.
     */
    customProperties: any;

    /**
     * Gets or sets id of the function.
     */
    id: string;

    /**
     * Gets value of a custom property.
     * @param name name of the custom property to get.
     * @returns value of the custom property or null if the property is undefined.
     */
    public getCustomProperty(name: string): any {
        if (this.customProperties != null && this.customProperties[name]) {
            return this.customProperties[name];
        }

        return null;
    }
}